src/main/java/org/apache/sling/commons/mime/internal/MimeTypeServiceImpl.java

```diff
@@ -166,9 +166,17 @@ public void registerMimeType(String mimeType, String... extensions) {
 
             }
         }
+        addToExtensions(defaultExtension, mimeType, extensions);
+    }
 
+    private void addToExtensions(String defaultExtension, String mimeType, String[] extensions) {
```

Comment:
I think the code would be easier to follow if you cleaned up the extensions array in the `registerMimeType` method ( lowercase, skip null/empty entries ) and no longer had to do so in `addToExtensions`.

I also think it's a good place to use `putIfAbsent` to simplify the code.

Reply:
Good suggestions, I have simplified the code